File: Preg1/fibonacci3-recursivo1.hs
import System.IO

fiboelem :: Integer -> Integer
fiboelem 1 = 0
fiboelem 2 = 0
fiboelem 3 = 1
fiboelem n = fiboelem ( n-1 ) + fiboelem ( n-2 ) + fiboelem ( n-3 )

main = do
    putStrLn "¿Cuantos elementos?: "
    x <- getLine
    let n = read x :: Integer
    print ([fiboelem(y)| y <- [1..n]])

File: Preg1/fibonacci3-recursivo1.scala
object Fibonacci {
    def fiboelem(n:Int) : Int =  {
        if (n<=1) then 0
        else if (n>3) then fiboelem(n-1) + fiboelem(n-2) + fiboelem (n-3)
        else  1
    }
    
    def fibo(n:Int) : Unit  =  {
        if(n>0) then {
            fibo(n-1)
            print (fiboelem(n) + " ")
        }
    }
    def main(args: Array[String]) = {
        println ("¿Cuantos elementos?: ")
        val x = scala.io.StdIn.readInt()
        fibo(x)
    }
} 